NearbyShare/SleepManager.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  SleepManager.swift
+//  NearbyShare
+//
+//  Created by Mika on 15.10.23.
+//
+
+import Foundation
+import IOKit
+import IOKit.pwr_mgt
+
+class SleepManager{
+    public static let shared=SleepManager()
+    private var assertionID: IOPMAssertionID=0
+    
+    public func disableSleep(){
+        if(assertionID != 0){
+            return
+        }
+
+        IOPMAssertionCreateWithName(
+            kIOPMAssertionTypePreventUserIdleSystemSleep as CFString,
+            IOPMAssertionLevel(kIOPMAssertionLevelOn),
+            "Data transfer over NearDrop" as CFString,
+            &assertionID
+        )
+    }
+    
+    public func enableSleep(){
+        if(assertionID == 0){
+            return
+        }
+        
+        IOPMAssertionRelease(assertionID)
+        assertionID = 0
+    }
```

Comment:
If there are multiple concurrent connections, and only one terminates, the assertion would get released anyway while the remaining connections are still active. This should be a reference counter.

Reply:
Because I am not really sure how to implement a reference counter, I am now looking for the number of active connections and if they are not 0 the sleep mode will not get reactivated. I hope you are fine with this approach.